internal/provider/networks_appliance_vlans_resource.go

```diff
@@ -0,0 +1,654 @@
+						CustomType:          jsontypes.BoolType,
+					},
+					"prefix_assignments": schema.SetNestedAttribute{
+						Optional:    true,
+						Computed:    true,
+						Description: "Prefix assignments on the VLAN",
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"autonomous": schema.BoolAttribute{
+									MarkdownDescription: "Auto assign a /64 prefix from the origin to the VLAN",
+									Optional:            true,
+									Computed:            true,
+									CustomType:          jsontypes.BoolType,
+								},
+								"static_prefix": schema.StringAttribute{
+									MarkdownDescription: "Manual configuration of a /64 prefix on the VLAN",
+									Optional:            true,
+									Computed:            true,
+									CustomType:          jsontypes.StringType,
+								},
+								"static_appliance_ip6": schema.StringAttribute{
+									MarkdownDescription: "Manual configuration of the IPv6 Appliance IP",
+									Optional:            true,
+									Computed:            true,
+									CustomType:          jsontypes.StringType,
+								},
+								"origin": schema.SingleNestedAttribute{
+									MarkdownDescription: "The origin of the prefix",
+									Optional:            true,
+									Computed:            true,
+									Attributes: map[string]schema.Attribute{
+										"type": schema.StringAttribute{
+											MarkdownDescription: "Type of the origin",
+											Optional:            true,
+											Computed:            true,
+											CustomType:          jsontypes.StringType,
+										},
+										"interfaces": schema.SetAttribute{
+											CustomType:  jsontypes.SetType[jsontypes.String](),
+											ElementType: jsontypes.StringType,
+											Description: "Interfaces associated with the prefix",
+											Computed:    true,
+											Optional:    true,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *NetworksApplianceVlansResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*openApiClient.APIClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *NetworksApplianceVlansResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *NetworksApplianceVlansResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	object56 := openApiClient.NewInlineObject56(data.Id.ValueString(), data.Name.ValueString())
```

Comment:
replace object56 to updateNetworkApplianceVlan

Reply:
done